js/customize-snapshots.js

```diff
@@ -139,1090 +156,104 @@
-				statusButton.hiddenButton.text( buttonText );
-				statusButton.buttonText.set( buttonText );
-			};
-
-			statusButton.value.bind( function( status ) {
-				selectedOption = statusButton.select.find( 'option:selected' );
-				statusButton.button.data( 'alt-text', selectedOption.data( 'alt-text' ) );
-				statusButton.button.data( 'button-text', selectedOption.text() );
-				statusButton.updateButtonText( 'button-text' );
-
-				if ( 'publish' === status ) {
-					snapshot.snapshotExpandButton.hide();
-					statusButton.button.data( 'confirm-text', selectedOption.data( 'confirm-text' ) );
-					statusButton.button.data( 'publish-text', selectedOption.data( 'publish-text' ) );
-					statusButton.needConfirm = true;
-				}
-
-				if ( 'future' === status ) {
-					snapshot.snapshotEditContainerDisplayed.set( true );
-					snapshot.snapshotExpandButton.show();
-					if ( snapshot.isFutureDate() ) {
-						snapshot.countdown.show();
-						snapshot.updateSnapshot( status );
-					}
-				} else {
-					snapshot.updateSnapshot( status );
-					snapshot.snapshotEditContainerDisplayed.set( false );
-					snapshot.countdown.hide();
+				toggleEditLinkControl: function() {
+					this.active.set( _.contains( [ 'draft', 'future', 'pending' ], api.state( 'changesetStatus' ).get() ) );
 				}
 			} );
 
-			statusButton.disbleButton.bind( function( disabled ) {
-				statusButton.button.prop( 'disabled', disabled );
-			} );
-
-			statusButton.disableSelect.bind( function( disabled ) {
-				statusButton.select.selectmenu( disabled ? 'disable' : 'enable' );
-				statusButton.dropDown.toggleClass( 'disabled', disabled );
-			} );
-
-			statusButton.disable = function( disable ) {
-				statusButton.disableSelect.set( disable );
-				statusButton.disbleButton.set( disable );
-			};
-
-			statusButton.button.on( 'click', function( event ) {
-				event.preventDefault();
-				snapshot.updateSnapshot( statusButton.value.get() );
-			} );
-
-			snapshot.publishButton.after( statusButton.container );
-
-			return statusButton;
+			api.control.add( new inspectLinkControl( 'inspect_changeset', {
+				type: 'inspect-changeset-link',
+				section: section.id,
+				priority: 30,
+				setting: api.state( 'changesetInspectLink' )
+			} ) );
 		},
 
 		/**
-		 * Remove 'customize_changeset_status' if it is being auto saved for edit box to avoid revisions.
+		 * Add pending status to changeset status control.
 		 *
 		 * @return {void}
 		 */
-		prefilterAjax: function prefilterAjax() {
-			var snapshot = this, removeParam, isSameStatus;
-
-			if ( ! api.state.has( 'changesetStatus' ) ) {
-				return;
-			}
-
-			removeParam = function( queryString, parameter ) {
-				var pars = queryString.split( /[&;]/g );
-
-				_.each( pars, function( string, index ) {
-					if ( string && string.lastIndexOf( parameter, 0 ) !== -1 ) {
-						pars.splice( index, 1 );
-					}
+		addPendingToStatusControl: function() {
+			var snapshot = this, params, coreStatusControl, index = 2;
```

Comment:
Instead of hard-coding the index of `2`, why not find the index dynamically? If it is supposed to be after `draft`, then find the draft and add one?